### ISSUE TITLE
Support Cowboy Update

### ISSUE THREAD

**Issue by k-cross** [handnot2/esaml#11]
Sorry if this is wrong, I come from the elixir side which is why I'm not submitting a PR and testing.

```
{deps, [
    cowboy,
    {cowboy, "1.1.2"},
    {cowboy, "2.*"}
]}.
```

**Comment by handnot2** [comment]
Published `esaml` v4.0.0. This release migrated to uptake Cowboy 2.x.
Anyone needing Cowboy 1.x should continue to use `esaml` v3.x.x.

**Comment by ribanez7** [comment]
Is there any way to use a wildcard to cover all 2.x versions? otherwise, can we update to cowboy 2.7? I can create the PR if needed. Thanks!
